apps/yapms/src/routes/app/+layout.svelte

```diff
@@ -1,50 +1,25 @@
 <script lang="ts">
+	import { page } from '$app/stores';
 	import '$lib/styles/global.css';
 	import '$lib/styles/roboto-swap.css';
-	import { goto } from '$app/navigation';
-	import { page } from '$app/stores';
-	import { get } from 'svelte/store';
-	import { LoadedMapStore } from '$lib/stores/LoadedMap';
-	import { SavedMapSchema } from '$lib/types/SavedMap';
 	import { onMount } from 'svelte';
-	import { LoadingErrorModalStore } from '$lib/stores/Modals';
-	import { PUBLIC_POCKETBASE_URI } from '$env/static/public';
+	import { get } from 'svelte/store';
+	import { loadPublicMap, loadUserMap } from '$lib/stores/LoadedMap';
+	import { goto } from '$app/navigation';
 
-	// this layout will will redirect the user to the proper map
-	// if they specify the m query parameter
 	onMount(async () => {
 		const url = get(page).url;
-		const map = encodeURIComponent(url.searchParams.get('m') ?? '');
-		// if the user is not loading a map, redirect them to the default map
-		if (map === '') {
-			if (url.pathname === '/app') {
-				await goto('/app/usa/presidential/2022');
-			}
-			return;
-		}
 
-		// load the requested map from pocketbase
-		console.log(`${PUBLIC_POCKETBASE_URI}/api/files/maps/${map}/data.json.gz`);
-		const data = await fetch(`${PUBLIC_POCKETBASE_URI}/api/files/maps/${map}/data.json.gz`);
-		const savedFile = SavedMapSchema.safeParse(await data.json());
+		const publicMapKey = url.searchParams.get('m');
+		const userMapKey = url.searchParams.get('um');
 
-		// if the saved map was malformed, redirect the user to the default map
-		if (!savedFile.success) {
-			LoadingErrorModalStore.set({
-				open: true
-			});
-			await goto('/app/usa/presidential/2022');
-			return;
+		if (publicMapKey !== null) {
+			await loadPublicMap(publicMapKey);
+		} else if (userMapKey !== null) {
+			await loadUserMap(userMapKey);
+		} else {
+			goto('/app/usa/presidential/2024');
```

Comment:
This goto will always run when you load a map that isn't saved. This layout file applies to all routes under /app so when you navigate to any map from the home page for example, you will be redirected. I fixed this on my end for testing by making this run only when `url.pathname === '/app'`